runtime/lua/vim/treesitter/query.lua

```diff
@@ -382,6 +382,36 @@ local predicate_handlers = {
 
     return string_set[node_text]
   end,
+
+  ['has-ancestor?'] = function(match, _, _, predicate)
+    local node = match[predicate[2]]
+    if not node then
+      return true
+    end
+
+    local ancestor_types = { unpack(predicate, 3) }
+    node = node:parent()
+    while node do
+      if vim.tbl_contains(ancestor_types, node:type()) then
+        return true
+      end
+      node = node:parent()
+    end
+    return false
+  end,
+
+  ['has-parent?'] = function(match, _, _, predicate)
+    local node = match[predicate[2]]
+    if not node then
+      return true
+    end
+
+    local ancestor_types = { unpack(predicate, 3) }
+    if vim.tbl_contains(ancestor_types, node:parent():type()) then
```

Comment:
Same remark as above.

Reply:
Not sure it's worth it here, as we're only doing this once (should be `list_contains`, though)?